pkg/proxy/endpoints.go

```diff
@@ -87,7 +81,7 @@ func (info *BaseEndpointInfo) GetIsLocal() bool {
 
 // IsReady returns true if an endpoint is ready and not terminating.
 func (info *BaseEndpointInfo) IsReady() bool {
-	return info.Ready
+	return info.Serving && !info.Terminating
```

Comment:
This is not true because of https://github.com/kubernetes/kubernetes/issues/116285